### PR TITLE
SystemTags for everything not just files

### PR DESCRIPTION
**Pull request by nickvergessen**
The logic is copied from the comments endpoint.

### CODE REVIEW

**Comment by LukasReschke** [comment]
👍

**Comment by schiessle** [comment]
:+1:

**Comment by mar1u5** [comment]
Conflicts :muscle: YAY :stuck_out_tongue_winking_eye:

File: apps/dav/lib/SystemTag/SystemTagsRelationsCollection.php [hunk 1]
```diff
@@ -56,10 +57,27 @@ public function __construct(
 				$tagMapper,
 				$userSession,
 				$groupManager,
-				$fileRoot
+				function($name) {
+					$nodes = \OC::$server->getUserFolder()->getById(intval($name));
```
Comment (icewind1991) [inline apps/dav/lib/SystemTag/SystemTagsRelationsCollection.php]: shouldn't the user folder be injected

File: apps/dav/lib/SystemTag/SystemTagsObjectTypeCollection.php [hunk 2]
```diff
@@ -76,27 +75,28 @@ class SystemTagsObjectTypeCollection implements ICollection {
...
+	 * @param \Closure $childExistsFunction
 	 */
 	public function __construct(
 		$objectType, 
 		ISystemTagManager $tagManager,
 		ISystemTagObjectMapper $tagMapper,
 		IUserSession $userSession,
 		IGroupManager $groupManager,
-		IRootFolder $fileRoot
+		\Closure $childExistsFunction
```
Comment (icewind1991) [inline apps/dav/lib/SystemTag/SystemTagsObjectTypeCollection.php]: I would prefer to inject something less generic (more "typed") than a closure, maybe something like a ``SystemTagsAdapter` that has a `chilldExists` method

If we do keep the closure than please use `callable` as type hint instead

Comment (nickvergessen) [inline apps/dav/lib/SystemTag/SystemTagsObjectTypeCollection.php]: I'd like to keep the API of comments (already released in 9) and tags similar here.
The problem is I didn't want to set up all the stuff, if we don't need it, that's why I made the fallback to closures, but yes, I also don't like this a lot.

**Pull request merged**